code/src/pages/product-details/components/product-details-container/ProductDetailsContainer.tsx

```diff
@@ -0,0 +1,154 @@
+import { useIntl } from "react-intl";
+
+import AppBadge from "@/components/app-badge/AppBadge";
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { deliveryMethods as deliveryMethodsData } from "@/constants/deliveryMethods";
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useErrorPageRedirect from "@/hooks/use-error-page-redirect/useErrorPageRedirect";
+import { productNotFoundRedirectConfig } from "@/pages/product-details/ProductsDetailsPage.constants";
+import { useGetUserProductByIdQuery } from "@/store/api/productsApi";
+import formatPrice from "@/utils/format-price/formatPrice";
+import getCategoryFromTags from "@/utils/get-category-from-tags/getCategoryFromTags";
+import isErrorWithStatus from "@/utils/is-error-with-status/isErrorWithStatus";
+
+import "@/pages/product-details/components/product-details-container/ProductDetailsContainer.scss";
+
+type ProductDetailsContainerProps = {
+  productId: string;
```

Comment:
Also, I suggest moving it to the products.types` and import here

Reply:
Thanks for review! I don't think that is is good idea to move local types to separate file, but I managed to reuse `ProductDetailsPageParams` type from previous comment

Reply:
I mean you can use in following:

type ProductDetailsContainerProps = {
  productId: ProductId; // wich is imported from `products.types`